__tests__/app.test.js

```diff
@@ -45,33 +45,73 @@ describe("/api", () => {
-          expect(response.body.article.topic).toBe('mitch');
-          expect(response.body.article.created_at).toBe('2020-11-03T09:12:00.000Z');
+          expect(response.body.article.body).toBe("some gifs");
+          expect(response.body.article.topic).toBe("mitch");
+          expect(response.body.article.created_at).toBe(
+            "2020-11-03T09:12:00.000Z"
+          );
           expect(response.body.article.votes).toBe(0);
-          expect(response.body.article.article_img_url).toBe('https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700');
+          expect(response.body.article.article_img_url).toBe(
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+          );
         });
     });
 
-    test('GET 404: sends an appropriate status and error message when given a valid id but does not exist', () => {
-        return request(app)
-        .get('/api/articles/100')
+    test("GET 404: sends an appropriate status and error message when given a valid id but does not exist", () => {
+      return request(app)
+        .get("/api/articles/100")
         .expect(404)
         .then((response) => {
-            expect(response.body.msg).toBe('Article not found')
-        })
-    })
+          expect(response.body.msg).toBe("Article not found");
+        });
+    });
 
-    test('GET 400: sends an appropriate status and error message when given an invalid id', () => {
-        return request(app)
-        .get('/api/articles/banana')
+    test("GET 400: sends an appropriate status and error message when given an invalid id", () => {
+      return request(app)
+        .get("/api/articles/banana")
         .expect(400)
         .then((response) => {
-            expect(response.body.msg).toBe('Bad request')
-        })
-    })
+          expect(response.body.msg).toBe("Bad request");
+        });
+    });
+  });
+  describe("GET /articles", () => {
+    test("GET 200: Should respond with an array of article objects each with the correct properties", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles.length).toBe(18);
+          expect(Array.isArray(articles)).toBe(true);
+          articles.forEach((article) => {
+            expect(typeof article.article_id).toBe("number");
+            expect(typeof article.title).toBe("string");
+            expect(typeof article.topic).toBe("string");
+            expect(typeof article.author).toBe("string");
+            expect(typeof article.created_at).toBe("string");
+            expect(typeof article.votes).toBe("number");
+            expect(typeof article.article_img_url).toBe("string");
+            expect(typeof article.comment_count).toBe("number");
+          });
+        });
+    });
```

Comment:
Good test Ahmed, this is structured correctly. Good work checking the array, if it's an array, array length and all properties including comment_count. Checking the 'shape' of the data here due to the number of objects coming back in the array is correct.